Faas docs (#1173)
Macrometacorp/docs: docs/photoniq/faas/faas-cli-install.md

---
sidebar_position: 20
title: FaaS CLI Installation
---

> This page explains how to install the PhotonIQ FaaS command line interface (CLI).

Reviewer: Change line to : Install the PhotonIQ FaaS Command Line Interface(CLI) to start creating and managing functions. Follow the installation instructions according to your Operating System(OS).
> [!NOTE]  
> To update the CLI with a newer version, follow the same installation procedure.

> This page explains how to install the PhotonIQ FaaS command line interface (CLI).

Reviewer 2: ```suggestion
Instructions for installing the PhotonIQ FaaS Command Line Interface (CLI).
```

> To update the CLI, follow the same procedure to install the new version over the old.

Reviewer 2: ```suggestion
To update the CLI, follow the installation steps again to replace the old version with the new one.
```

Refer to the section that matches your operating system.

> ### MacOS

Reviewer: ### Install on MacOS

> 1) Install the Rust compiler and WebAssembly libraries by executing the following commands. This is required for creating Rust functions:

Reviewer: I'd suggest turning this first step into a prerequisite. Before the MacOS installation step, create a subheading - Prerequisites and list this step before installation. Like this:
Installation on MacOS (h2)
Prerequisites (h3)
list step here
Steps(h3)
List other steps here

```shell
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
rustup target add wasm32-wasi
```

> 2) Download the FaaS CLI package to your machine from link below:

Reviewer: Turn this line to : Download the [FaaS MacOS CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-apple-darwin.tar.gz) on your local machine.

> [FaaS MacOS CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-apple-darwin.tar.gz)

Reviewer 2: Does this work on M(1,2,3,4) machines?

3) Navigate into the CLI directory, and then launch the `faas` CLI tool:

```shell
./faas help
```

> You can now run [FaaS CLI Commands](faas-cli-commands.md).

Reviewer: Turn this into a tip alert like this:
> [!TIP]
> Get familiar with the FaaS CLI by practicing with some of our [available commands](faas-cli-commands.mdx).

### Windows

***Prerequisites:***

> Install Microsoft Build Tools [Download C++ build tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/)

Reviewer: ### Prerequisites
1. Install [Microsoft Build Tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/)
2. To compile Rust functions, select **Desktop development with C++** from the installation menu. 
3. Install WebAssembly libraries by executing the following command. This enables you to create Rust functions:

In the installation menu select Desktop development with C++ (This is enough to compile Rust functions)

1) Install WebAssembly libraries by executing the following command. This is required for creating Rust functions:

```shell
rustup target add wasm32-wasi
```

2) Download the FaaS CLI package to your machine from link below:

> [FaaS Windows CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-pc-windows-gnu.zip)

Reviewer: Repeat the same correction.: Download the [FaaS MacOS CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-apple-darwin.tar.gz) on your local machine.

Reviewer: Before the download step, precede it with a Steps heading. Like:
### Steps

> 3) Navigate into the CLI directory, and then launch the `faas` CLI tool

Reviewer: Navigate into the CLI directory and launch the `faas` CLI tool

```shell
./faas help
```

> You can now run [FaaS CLI Commands](faas-cli-commands.md).

Reviewer: Turn to 'Now you can start creating and deploying your first function.
Note: **Deploying your first function** is a tutorial I'm working on, so it's supposed to be a link.

### Linux

1) Install the Rust compiler and WebAssembly libraries by executing the following commands. This is required for creating Rust functions:

```shell
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
rustup target add wasm32-wasi
```

> 2) Download the FaaS CLI package to your machine from link below:

Reviewer: Change to: Depending on your Linux system architecture, you can download the [FaaS Linux GNU CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-aarch64-unknown-linux-gnu.tar.gz) or [FaaS Linux MUSL CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-unknown-linux-musl.tar.gz).

[FaaS Linux GNU CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-aarch64-unknown-linux-gnu.tar.gz)

[FaaS Linux MUSL CLI package](https://macrometacorp.github.io/photoniq-faas-cli-docs/faas-1.0.0-x86_64-unknown-linux-musl.tar.gz)

3) Navigate into the CLI directory, and then launch the `faas` CLI tool

```shell
./faas help
```

> You can now run [FaaS CLI Commands](faas-cli-commands.md).

Reviewer: Remove this line
